File: julia-package/brainflow/test/release_all.jl
using BrainFlow

# enable all possible logs from all three libs
BrainFlow.enable_dev_logger(BrainFlow.BOARD_CONTROLLER)
BrainFlow.enable_dev_logger(BrainFlow.DATA_HANDLER)
BrainFlow.enable_dev_logger(BrainFlow.ML_MODULE)

params1 = BrainFlowInputParams()
params1.other_info = "board1"
board_shim1 = BrainFlow.BoardShim(BrainFlow.SYNTHETIC_BOARD, params1)
params2 = BrainFlowInputParams()
params2.other_info = "board2"
board_shim2 = BrainFlow.BoardShim(BrainFlow.SYNTHETIC_BOARD, params2)

BrainFlow.prepare_session(board_shim1)
BrainFlow.prepare_session(board_shim2)
BrainFlow.start_stream(board_shim1)
sleep(2)
data = BrainFlow.get_current_board_data(10, board_shim1)
println(BrainFlow.calc_stddev(data[2, :]))

BrainFlow.release_all_sessions()

# calc concentration
model_params = BrainFlowModelParams(BrainFlow.RESTFULNESS, BrainFlow.DEFAULT_CLASSIFIER)
BrainFlow.prepare(model_params)
BrainFlow.release_all()

File: tests/python/release_all.py
from brainflow.board_shim import BoardShim, BrainFlowInputParams, BoardIds
from brainflow.ml_model import MLModel, BrainFlowMetrics, BrainFlowClassifiers, BrainFlowModelParams


def main():
    BoardShim.enable_dev_board_logger()

    params1 = BrainFlowInputParams()
    params1.other_info = 'board1'
    board_id = BoardIds.SYNTHETIC_BOARD.value
    params2 = BrainFlowInputParams()
    params2.other_info = 'board2'

    board1 = BoardShim(board_id, params1)
    board1.prepare_session()
    board2 = BoardShim(board_id, params2)
    board2.prepare_session()

    BoardShim.release_all_sessions()

    ml_params1 = BrainFlowModelParams(BrainFlowMetrics.RESTFULNESS.value, BrainFlowClassifiers.DEFAULT_CLASSIFIER.value)
    model1 = MLModel(ml_params1)
    model1.prepare()
    ml_params2 = BrainFlowModelParams(BrainFlowMetrics.MINDFULNESS.value, BrainFlowClassifiers.DEFAULT_CLASSIFIER.value)
    model2 = MLModel(ml_params2)
    model2.prepare()

    MLModel.release_all()


if __name__ == "__main__":
    main()
